packages/service/ni_measurement_plugin_sdk_service/_internal/parameter/decoder.py

```diff
@@ -57,14 +51,18 @@ def deserialize_parameters(
     return parameter_values
 
 
-def _deserialize_enum_parameter(
-    repeated: bool, field_value: Any, enum_type: Any
-) -> Union[List[Any], Any]:
+def _deserialize_enum_parameter(field_value: Any, metadata: ParameterMetadata) -> Any:
     """Convert all enums into the user defined enum type.
 
     Returns:
         Union[List[Any], Any]: Enum type or a list of enum types.
     """
-    if repeated:
-        return [enum_type(value) for value in field_value]
-    return enum_type(field_value)
+    try:
+        # ValueType is defined when field_value is a protobuf enum
+        metadata.enum_type.ValueType
+        return field_value
+    except AttributeError:
+        enum_type = metadata.enum_type
+        if metadata.repeated:
+            return [enum_type(value) for value in field_value]
+        return enum_type(field_value)
```

Comment:
Here's how to do this without throwing an exception.
```suggestion
    if hasattr(metadata.enum_type, "ValueType"):
        # ValueType is defined when field_value is a protobuf enum
        metadata.enum_type.ValueType
        return field_value
    else:
        enum_type = metadata.enum_type
        if metadata.repeated:
            return [enum_type(value) for value in field_value]
        return enum_type(field_value)
```